pkg/metrics/nova.go

```diff
@@ -128,7 +129,20 @@ func PublishServerMetrics(client *gophercloud.ServiceClient, tenantID string) er
 
 // publishServerMetric extracts data from a server and exposes the metrics via prometheus
 func publishServerMetric(srv servers.Server) {
-	labels := []string{srv.ID, srv.Name}
+
+	// Pick up from OpenStack metadata
+	environment := ""
+	if env, exists := srv.Metadata["env"]; exists {
+		environment = env
+	}
+
+	// We could also use the naming to detect env.
+	environmentFromName := ""
+	if strings.Contains(srv.Name, "test-2") {
```

Comment:
I don't get this `test-2` string. What do you want to do with this?

Reply:
Sorry so this was a long time ago and (one of) my first pr so it's pretty unclear what the idea is or if it's useful. If I recall correctly the things was we may have multiple clusters for different purposes and the environment is in the name. E g `foo-test-1` or `foo-test-2`. The idea was to provide kosmoo with a list of environments to look for `[test-1, test-2]` and then check if the name contains one of environments and if so use it.